addons/stock/models/stock_quant.py

```diff
@@ -221,17 +220,118 @@ def action_view_stock_moves(self):
+            if (quant.inventory_quantity - quant.inventory_diff_quantity) != quant.quantity:
+                quants_outdated.append(quant.id)
+            if quant.product_id.tracking in ['lot', 'serial'] and\
+                    not quant.lot_id and quant.inventory_quantity != quant.quantity:
+                products_tracked_without_lot.append(quant.product_id.id)
+        # for some reason if multi-record, env.context doesn't pass to wizards...
+        ctx = dict(self.env.context or {})
+        ctx['default_quant_ids'] = self.ids
+        if quants_outdated:
+            ctx['default_quant_to_fix_ids'] = quants_outdated
+            return {
+                'name': _('Conflict in Inventory Adjustment'),
+                'type': 'ir.actions.act_window',
+                'view_mode': 'form',
+                'views': [(False, 'form')],
+                'res_model': 'stock.inventory.conflict',
+                'target': 'new',
+                'context': ctx,
+            }
+        if products_tracked_without_lot:
+            ctx['default_product_ids'] = products_tracked_without_lot
+            return {
+                'name': _('Tracked Products in Inventory Adjustment'),
+                'type': 'ir.actions.act_window',
+                'view_mode': 'form',
+                'views': [(False, 'form')],
+                'res_model': 'stock.track.confirmation',
+                'target': 'new',
+                'context': ctx,
+            }
+        self._apply_inventory()
+
+    def action_inventory_history(self):
+        self.ensure_one()
+        action = {
+            'name': _('History'),
+            'view_type': 'tree',
+            'view_mode': 'list,form',
+            'res_model': 'stock.move.line',
+            'views': [(self.env.ref('stock.view_move_line_tree').id, 'list'), (False, 'form')],
+            'type': 'ir.actions.act_window',
+            'context': {
+                'search_default_inventory': 1,
+                'search_default_done': 1,
+            },
+            'domain': [
+                ('product_id', '=', self.product_id.id),
+                ('company_id', '=', self.company_id.id),
+                '|',
+                    ('location_id', '=', self.location_id.id),
+                    ('location_dest_id', '=', self.location_id.id),
+            ],
+        }
+        if self.lot_id:
+            action['context']['search_default_lot_id'] = self.lot_id.id
+        if self.package_id:
+            action['context']['search_default_package_id'] = self.package_id.id
+            action['context']['search_default_result_package_id'] = self.package_id.id
+        if self.owner_id:
+            action['context']['search_default_owner_id'] = self.owner_id.id
+        return action
+
+    def action_set_inventory_quantity(self):
+        for quant in self:
+            quant.inventory_quantity = quant.quantity
+
+    def action_set_inventory_quantity_to_zero(self):
+        for quant in self:
```

Comment:
IMO, No need to loop here, you can batch the assign.

Reply:
Fixed. There's currently no way to apply it to multiple quants anyways.